channeldb/migtest/migtest.go

```diff
@@ -0,0 +1,94 @@
+package migtest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/coreos/bbolt"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/prometheus/common/log"
+)
+
+// MakeDB creates a new instance of the ChannelDB for testing purposes. A
+// callback which cleans up the created temporary directories is also returned
+// and intended to be executed after the test completes.
+func MakeDB() (*channeldb.DB, func(), error) {
+	// First, create a temporary directory to be used for the duration of
+	// this test.
+	tempDirName, err := ioutil.TempDir("", "channeldb")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Next, create channeldb for the first time.
+	cdb, err := channeldb.Open(tempDirName)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cleanUp := func() {
+		cdb.Close()
+		os.RemoveAll(tempDirName)
+	}
+
+	return cdb, cleanUp, nil
+}
+
+// ApplyMigration is a helper test function that encapsulates the general steps
+// which are needed to properly check the result of applying migration function.
+func ApplyMigration(t *testing.T,
```

Comment:
Ref this in `migration_01_to_11`? Maybe not...

Reply:
in that package the after and before functions operate on `channeldb.DB`, where as here they operate on `bbolt.Tx`. some of those functions rely on db-level methods, so it is a bigger refactor. i made the decision here to only offer `bbolt.Tx` because it makes the closures more isolated, and hopefully less code to copy in future ones.

Reply:
sounds good